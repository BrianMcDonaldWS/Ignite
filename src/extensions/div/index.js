import container from 'markdown-it-container';
import parseClasses from '../utils/parse-classes';

const regExp = /div/;

const div = {
  validate(params) {
    return params.trim().match(regExp);
  },

  render(tokens, idx) {
    const classList = parseClasses(tokens[idx].info);
    const classAttr =
      (tokens[idx].attrs || []).find(([key]) => key === 'class') || [];
    console.log(`${classAttr[1]} ${classList.join(' ')}`);
    if (tokens[idx].nesting === 1) {
      return `
          <div class="${classAttr[1]} ${classList.join(' ')}">
      `;
    }

    return `
      </div>
    `;
  }
};

export default function bulmaMessage(md) {
  md.use(container, 'div', div);
}
