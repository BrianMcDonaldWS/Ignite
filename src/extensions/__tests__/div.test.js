import MarkdownIt from 'markdown-it';
import attrs from 'markdown-it-attrs';
import div from '../div';

const md = new MarkdownIt().use(div).use(attrs);

test('div', () => {
  expect(
    md.render(`
::: div with some classes
content
:::
  `)
  ).toMatchSnapshot();

  expect(
    md.render(`
::: div {.class .names}
content
:::
  `)
  ).toMatchSnapshot();
});
