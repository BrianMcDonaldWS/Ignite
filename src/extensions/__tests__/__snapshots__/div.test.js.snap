// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`div 1`] = `
"
          <div class=\\"undefined with some classes\\">
      <p>content</p>

      </div>
    "
`;

exports[`div 2`] = `
"
          <div class=\\"class names \\">
      <p>content</p>

      </div>
    "
`;
